src/components/Footer.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+export default class Footer extends React.Component {
+    buttons = [
+        {name: 'all', label: 'All'},
+        {name: 'active', label: 'Active'},
+        {name: 'done', label: 'Done'}
+    ];
+    render() {
+        const {onFilterChange} = this.props;
+        let index = 0;
+        let done = this.props.done;
+        let clearCompleted;
+        if(done){
+             clearCompleted = (
+                <button
+                    className="clear-completed"
+                    onClick={() => this.props.deleteAll()}>
+                    Clear completed
+                </button>)
+        } else {
+            clearCompleted = null;
+        }
+        const buttons = this.buttons.map(({name, label}) => {
+            return (
+                <li key={++index}>
```

Comment:
Avoid every kind of `index` for `key` prop. You can add `id` to your buttons array